internal/nginx/version.go

```diff
@@ -0,0 +1,100 @@
+package nginx
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+// Version holds the parsed output from `nginx -v`.
+type Version struct {
+	Raw    string
+	OSS    string
+	IsPlus bool
+	Plus   string
+}
+
+// NewVersion will take the output from `nginx -v` and explodes it into the `nginx.Version` struct.
+func NewVersion(line string) Version {
+	matches := re.FindStringSubmatch(line)
+	plusmatches := plusre.FindStringSubmatch(line)
+	nv := Version{
+		Raw: line,
+	}
+
+	if len(plusmatches) > 0 {
+		subNames := plusre.SubexpNames()
+		nv.IsPlus = true
+		for i, v := range plusmatches {
+			switch subNames[i] {
+			case "plus":
+				nv.Plus = v
+			case "version":
+				nv.OSS = v
+			}
+		}
+	}
+
+	if len(matches) > 0 {
+		for i, key := range re.SubexpNames() {
+			val := matches[i]
+			if key == "version" {
+				nv.OSS = val
+			}
+		}
+	}
+
+	return nv
+}
+
+// String returns the raw Nginx version string from `nginx -v`.
+func (v Version) String() string {
+	return v.Raw
+}
+
+// Format returns a string representing Nginx version.
+func (v Version) Format() string {
```

Comment:
This is to be called in `main.go` ?

Reply:
Yes, will be dressed in a separate PR

> This is to be called in `main.go` ?

Yes, will be dressed in a separate PR